tensordict/tensordict.py

```diff
@@ -1580,7 +1585,9 @@ def split(
                 "split(): argument 'split_size' must be int or list of ints"
             )
         dictionaries = [{} for _ in range(len(batch_sizes))]
-        for key, item in self.items():
+        key_view = _TensorDictKeysView(self, include_nested=True, leaves_only=False, error_on_loop=False)
+        for key in key_view:
+            item = self.get(key)
```

Comment:
This doesn't give the right answer either. I don't think we can use the keys view in cases like this. We can replace if there is an existing `.keys(include_nested=True)` or similar, or if we want to loop over all values and perform some in-place operation on them (see `zero_`) / perform some check (see `any` and `all`). But if `include_nested=False` and we change it to `True`, the things we're iterating over change quite substantially.

I think this would be another case for `_apply_safe`

Reply:
Corrected following your point